File: dsa-in-javascript/Container-With-Most-Water.js
function maxArea(arr)
{
    let left = 0;
    let right = arr.length-1;
    let maxarea = 0;

    while(left < right){
        let ht =  Math.min(arr[left], arr[right]);
         
        maxarea = Math.max(maxarea, ht * (right-left));
       

        if(arr[left] < arr[right]){
            left++;
        }else{
            right--;
        }
    }
     
    return maxarea;

}

const arr = [1,8,6,2,5,4,8,3,7];
console.log(maxArea(arr));

File: dsa-in-golang/Container-With-Most-Water.go
package main

import (
	"fmt"
	"math"
)

func maxArea(height []int) int {
	left := 0
	right := len(height) - 1
	maxarea := 0

	for left < right {
		ht := float64(min(height[left], height[right]))
		maxarea = int(math.Max(float64(maxarea), ht*float64(right-left)))

		if height[left] < height[right] {
			left++
		} else {
			right--
		}

	}

	return maxarea
}

func main() {
	height := []int{1, 8, 6, 2, 5, 4, 8, 3, 7}
	result := maxArea(height)
	fmt.Println(result)
}
